src/klein/_tubes.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2017-2018. See LICENSE for details.
+
+"""
+Extensions to Tubes.
+"""
+
+from io import BytesIO
+from typing import Iterable
+from typing.io import BinaryIO
+
+from attr import attrib, attrs
+from attr.validators import instance_of, optional, provides
+
+from tubes.itube import IDrain, IFount, ISegment
+from tubes.kit import Pauser, beginFlowingTo
+from tubes.undefer import fountToDeferred
+
+from twisted.internet.defer import Deferred
+from twisted.python.failure import Failure
+
+from zope.interface import implementer
+
+BinaryIO, Deferred, Iterable  # Silence linter
+
+
+__all__ = ()
+
+
+# See https://github.com/twisted/tubes/issues/60
+def fountToBytes(fount):
+    # type: (IFount) -> Deferred[bytes]
+    def collect(chunks):
+        # type: (Iterable[bytes]) -> bytes
+        return b"".join(chunks)
+
+    d = fountToDeferred(fount)
+    d.addCallback(collect)
+    return d
+
+
+# See https://github.com/twisted/tubes/issues/60
+def bytesToFount(data):
+    # type: (bytes) -> IFount
+    return IOFount(source=BytesIO(data))
+
+
+
+# https://github.com/twisted/tubes/issues/61
+@implementer(IFount)
+@attrs(frozen=False)
+class IOFount(object):
+    """
+    Fount that reads from a file-like-object.
+    """
+
+    outputType = ISegment
+
+    _source = attrib()  # type: BinaryIO
+
+    drain = attrib(
+        validator=optional(provides(IDrain)), default=None, init=False
```

Comment:
why not have a throw-away drain and have that as the default?

Reply:
Well, for one thing that's more code, so more complexity… I think that's probably a good idea when this get adopted into `tubes`, where it belongs, but `IOFount` isn't intended to be exported by Klein; it's just used internally to get us to a point where we can have a working API.  That is to say that `IOFount` definitely needs more work.